app/lib/package.rb

```diff
@@ -52,6 +55,11 @@ def determine_resources(document)
     nodes.each { |node| @resources << node.content }
   end
 
+  def determine_tips(document)
+    nodes = XmlUtil.xpath_namespace(document, XPATH_TIPS.join("|"))
+    nodes.each { |node| @tips << Tip.find_by(name: node.content) }
```

Comment:
what happens if the tip isn't found by this statement? we want to raise an error in that situation

Reply:
also, does this only find tips within the current resource?